src/Compilers/CSharp/Test/Syntax/Parsing/LineSpanDirectiveParsingTests.cs

```diff
@@ -0,0 +1,1585 @@
+
+            N(SyntaxKind.LineSpanDirectiveTrivia);
+            {
+                N(SyntaxKind.HashToken);
+                N(SyntaxKind.LineKeyword);
+                N(SyntaxKind.LineDirectivePosition);
+                {
+                    N(SyntaxKind.OpenParenToken);
+                    N(SyntaxKind.NumericLiteralToken, "1");
+                    N(SyntaxKind.CommaToken);
+                    N(SyntaxKind.NumericLiteralToken, "2");
+                    N(SyntaxKind.CloseParenToken);
+                }
+                N(SyntaxKind.MinusToken);
+                M(SyntaxKind.LineDirectivePosition);
+                {
+                    M(SyntaxKind.OpenParenToken);
+                    M(SyntaxKind.NumericLiteralToken);
+                    M(SyntaxKind.CommaToken);
+                    M(SyntaxKind.NumericLiteralToken);
+                    M(SyntaxKind.CloseParenToken);
+                }
+                M(SyntaxKind.StringLiteralToken);
+                N(SyntaxKind.EndOfDirectiveToken);
+            }
+            EOF();
+        }
+
+        [Fact]
+        public void Incomplete_07()
+        {
+            string source = @"#line (1, 2) - (";
+
+            UsingLineDirective(source, options: null,
+                // (1,17): error CS1576: The line number specified for #line directive is missing or invalid
+                // #line (1, 2) - (
+                Diagnostic(ErrorCode.ERR_InvalidLineNumber, "").WithLocation(1, 17));
+
+            N(SyntaxKind.LineSpanDirectiveTrivia);
+            {
+                N(SyntaxKind.HashToken);
+                N(SyntaxKind.LineKeyword);
+                N(SyntaxKind.LineDirectivePosition);
+                {
+                    N(SyntaxKind.OpenParenToken);
+                    N(SyntaxKind.NumericLiteralToken, "1");
+                    N(SyntaxKind.CommaToken);
+                    N(SyntaxKind.NumericLiteralToken, "2");
+                    N(SyntaxKind.CloseParenToken);
+                }
+                N(SyntaxKind.MinusToken);
+                N(SyntaxKind.LineDirectivePosition);
+                {
+                    N(SyntaxKind.OpenParenToken);
+                    M(SyntaxKind.NumericLiteralToken);
+                    M(SyntaxKind.CommaToken);
+                    M(SyntaxKind.NumericLiteralToken);
+                    M(SyntaxKind.CloseParenToken);
+                }
+                M(SyntaxKind.StringLiteralToken);
+                N(SyntaxKind.EndOfDirectiveToken);
+            }
+            EOF();
+        }
+
+        [Fact]
+        public void Incomplete_08()
+        {
+            string source = @"#line (1, 2) - (3";
+
+            UsingLineDirective(source, options: null,
+                // (1,18): error CS1003: Syntax error, ',' expected
+                // #line (1, 2) - (3
+                Diagnostic(ErrorCode.ERR_SyntaxError, "").WithArguments(",", "").WithLocation(1, 18));
```

Comment:
I think it's almost a matter of style, but--in my view, it's not necessary to limit the errors given to just the "next" error. If a user is in this position, they will have to add a ',', then a number, then a ')', then possibly an offset, and a filename, etc. before the directive is valid. As long as the diagnostics for each of the missing pieces are given in the appropriate sequence it seems fine and maybe good to give all of them since it's gonna reduce the need to wait for that round-trip to find out what the next thing they need to fix is.

Reply:
Granted, if you made that change, you might want to ensure that "extra/unexpected" tokens were attached to the last good token, and seek ahead to look for the next "expected" token, so that in a test like 'UnexpectedToken_08' you don't give a bunch of diagnostics for "expected such-and-such token" for tokens that are actually there--instead you just give a diagnostic for the unexpected 'u'.

Reply:
> it's not necessary to limit the errors given to just the "next" error

I agree, and in general I prefer seeing all errors, but for these cases of syntax errors, it wasn't clear how to best recover from a missing or unexpected token, other than to skip to the end of the directive.